src/opnsense/www/js/widgets/IpsecLeases.js

```diff
@@ -0,0 +1,145 @@
+        $('#ipsecLeaseTable').empty().append($error);
+    }
+
+    // Function to process leases data and update the UI accordingly
+    processLeases(data) {
+        let users = {}; // Initialize an object to store user data indexed by user names
+
+        // Organize leases by user
+        data.leases.forEach(lease => {
+            if (!users[lease.user]) {
+                users[lease.user] = {
+                    ipAddresses: [], // List of IP addresses assigned to the user
+                    online: false    // Online status initially false
+                };
+            }
+            users[lease.user].ipAddresses.push({ address: lease.address, online: lease.online });
+            // Set the user's status to online if any of their IP addresses is online
+            if (lease.online) {
+                users[lease.user].online = true;
+            }
+        });
+
+        // Sort users by online status, placing online users first
+        let sortedUsers = Object.keys(users).sort((a, b) => {
+            return users[a].online === users[b].online ? 0 : users[a].online ? -1 : 1;
+        });
+
+        // Calculate and display the number of users currently online and total users
+        let onlineUsersCount = Object.values(users).filter(user => user.online).length;
+        let totalUsersCount = Object.keys(users).length;
+        let offlineUsersCount = totalUsersCount - onlineUsersCount;
+
+        // Prepare a summary row for user counts
+        let userCountsRow = `
+            <div>
+                <span><b>${this.translations.users}:</b> ${totalUsersCount} - <b>${this.translations.online}:</b> ${onlineUsersCount} - <b>${this.translations.offline}:</b> ${offlineUsersCount}</span>
+            </div>`;
+
+        let rows = [userCountsRow];
+        // Prepare HTML content for each user showing their status and IP addresses
+        sortedUsers.forEach(user => { // Use sortedUsers instead of Object.keys(users)
+            let userStatusClass = users[user].online ? 'text-success' : 'text-danger';
+            let userStatusTitle = users[user].online ? this.translations.online : this.translations.offline;
+
+            let row = `
+                <div>
+                    <i class="fa fa-user ${userStatusClass}" style="cursor: pointer;"
+                        data-toggle="tooltip" title="${userStatusTitle}">
+                    </i>
+                    &nbsp;
+                    <span><b>${user}</b></span>
+                    <br/>
+                    <div style="margin-top: 5px; margin-bottom: 5px;">
+                        ${users[user].ipAddresses.map(ip => `<div>${ip.address}</div>`).join('')}
+                    </div>
+                </div>`;
+            rows.push(row);
+        });
+
+        // Update the HTML table with the sorted rows
+        super.updateTable('ipsecLeaseTable', rows.map(row => [row]));
```

Comment:
Thanks for the feedback. I'll try to improve that.

Reply:
@swhite2 I'm not sure I'm able to only update specific rows, since I am sorting based on online/offline status. When a status changes, the order would also need to change, updating all rows always makes the order of elements work. I'd need some help here if it needs to become more intricate.